sentry_sdk/tracing.py

```diff
@@ -1272,15 +1276,42 @@ def __init__(
         listed in the signature. These additional arguments are ignored.
         """
         from sentry_sdk.integrations.opentelemetry.consts import SentrySpanAttribute
+        from sentry_sdk.integrations.opentelemetry.utils import (
+            convert_to_otel_timestamp,
+        )
+
+        if start_timestamp is not None:
+            # OTel timestamps have nanosecond precision
+            start_timestamp = convert_to_otel_timestamp(start_timestamp)
 
-        self._otel_span = tracer.start_span(description or op or "")  # XXX
+        # XXX deal with _otel_span being a NonRecordingSpan
+        self._otel_span = tracer.start_span(
+            description or op or "", start_time=start_timestamp
+        )  # XXX
         self._active = active
 
         self._otel_span.set_attribute(SentrySpanAttribute.ORIGIN, origin)
         if op is not None:
-            self._otel_span.set_attribute(SentrySpanAttribute.OP, op)
-        if description is not None:
-            self._otel_span.set_attribute(SentrySpanAttribute.DESCRIPTION, description)
+            self.op = op
+        self.description = description
```

Comment:
Do you mean the `is not None` check that's there for `op` but not `description`?

This should probably be unified. I wasn't sure about the semantics of setting `op` or `description` to `None` (would anyone want to do that?). More importantly, [OTel doesn't really support `None` as a valid value](https://github.com/open-telemetry/opentelemetry-python/blob/7291af2e63c9bbd029e52d080c93614796f78822/opentelemetry-api/src/opentelemetry/util/types.py#L18) for span attributes (see also [this comment](https://github.com/open-telemetry/opentelemetry-python/blob/7291af2e63c9bbd029e52d080c93614796f78822/opentelemetry-api/src/opentelemetry/trace/span.py#L101)).

So I guess the correct way to go about this would be:
- if an attribute like `op` or `description` is set to `None`: do nothing (deleting attributes is not really a thing in OTel as far as I can tell)
- if it's set to something non-`None`, set the attr

So I'd move this logic to check whether the value is not `None` to the actual properties for all attrs where it makes sense instead of having it here in the `__init__`.

Reply:
https://github.com/getsentry/sentry-python/pull/3441/commits/8c0d3338f5fd2b910a5bdc02f8b66ff0ba4841d4